8368514

djelinski opened the pull request:
Please review this trivial patch that fixes stateless session resumption with JCE providers that require extra space for AES/GCM decryption.

I modified the existing FipsModeTLS12 test to additionally verify that session resumption works. The TLS 1.3 test resumes the session using a stateless ticket; the TLS 1.2 test uses stateful sessions, because stateless ticket creation fails for other reasons.

Tier1-3 tests continue to pass.

artur-oracle commented on test/jdk/sun/security/pkcs11/tls/tls12/FipsModeTLS12.java:
```diff
@@ -272,12 +274,14 @@ private static void testTlsAuthenticationCodeGeneration()
                 SSLEngine clientSSLEngine = engineToTest[0];
                 SSLEngine serverSSLEngine = engineToTest[1];
-
+                // Check that session resumption works
```
I would put more details in this comment about what exactly we are testing here: stateless session ticket decryption using the most preferred currently installed provider - NSS.

artur-oracle commented on test/jdk/sun/security/pkcs11/tls/tls12/FipsModeTLS12.java:
```diff
@@ -24,14 +24,15 @@
  * @run main/othervm/timeout=120 -Djdk.tls.client.protocols=TLSv1.2
- *      -Djdk.tls.useExtendedMasterSecret=false FipsModeTLS12
+ *      -Djdk.tls.useExtendedMasterSecret=false
+ *      -Djdk.tls.client.enableSessionTicketExtension=false FipsModeTLS12
```
Why do we need to set `jdk.tls.client.enableSessionTicketExtension` to false?

djelinski replied:
because with TLS 1.2, the server can't produce a stateless ticket; it fails when trying to extract the master secret.

djelinski commented:
Thanks @artur-oracle @valeriepeng for the reviews. I added more comments and renamed the test to stop referring to TLS 1.2. Let me know if that addresses your concerns.

artur-oracle commented:
> Thanks @artur-oracle @valeriepeng for the reviews. I added more comments and renamed the test to stop referring to TLS 1.2. Let me know if that addresses your concerns.

Thanks for adding more comments and filing JDK-8368669!

valeriepeng reviewed (approved):
Changes look good.

djelinski commented:
Thanks @artur-oracle @valeriepeng for the re-reviews.

/integrate
